src/uct/ib/base/ib_md.c

```diff
@@ -1418,7 +1418,10 @@ static double uct_ib_md_pci_bw(const uct_ib_md_config_t *md_config,
 
     for (i = 0; i < md_config->pci_bw.count; i++) {
         if (!strcmp(ib_device->name, md_config->pci_bw.device[i].name)) {
-            return md_config->pci_bw.device[i].bw * 0.9846; /* ~1.54% lost for PCI Gen3 */
+            return md_config->pci_bw.device[i].bw * 0.9846; /* ~1.54% lost for PCI Gen3. See below */
+           /* PCI Express 3.0 uses 128b/130b encoding scheme: 128 bits of data + 2 bits for *
+            * correctness. It means that ~1.54% (1 - 128/130) of data is used for PCI       *
```

Comment:
this calculation is incomplete, for 8x Gen3 should be ~108 Gb/s
also - what about Gen1/Gen2?

Reply:
pls do it like this: https://github.com/openucx/ucx/blob/95031af7e4943411b7d23327db0b24bc9c11908c/src/uct/ib/base/ib_iface.c#L1163

Reply:
Gen1/2 uses 8/10 bits encoding, about 20% overhead, and there is no way to know which gen is used (assuming the most actual for now - Gen3)

are you sure that 8x Gen3 (8Gbs) provides 108 Gbs? not 16x?
note that real speed will depends from memory/PCI/HCA protocol which is magic number...

ok, updated

Reply:
we can detect gen according to speed of single lane: Gen1: 2.5Gbs, Gen2: 5Gbs, Gen3: 8Gbs
in PCIe there is encoding and above that TLP header + flow control overhead
see https://www.xilinx.com/support/documentation/white_papers/wp350.pdf

Reply:
according offline discussion - multiplexer is removed